spec/refresher_spec.rb

```diff
@@ -5,78 +5,78 @@
-      expect(Array.new.instance_of? _FILL_ME_IN_).to be true
+      expect([].is_a? Array).to be true
+      expect(Array.new.instance_of? Array).to be true
     end
 
     it "Yes, it is a Object, but which class" do
-      expect({}.is_a? _fill_in_object_).to be true
-      expect(Hash.new.instance_of? _FILL_ME_IN_).to be true
+      expect({}.is_a? Hash).to be true
+      expect(Hash.new.instance_of? Hash).to be true
     end
   end
 
   context "guess the type" do
     it "Yes, it is a Object, but which" do
-      expect(:class.is_a? _fill_in_object_).to be true
+      expect(:class.is_a? Symbol).to be true
     end
 
     it "Yes, it is a Object, but which" do
-      expect("module".is_a? _fill_in_object_).to be true
+      expect("module".is_a? String).to be true
     end
 
     it "Yes, it is a Object, but which" do
-      expect(1.is_a? _fill_in_object_).to be true
+      expect(1.is_a? Integer).to be true
     end
 
     it "Yes, it is a Object, but which" do
-      expect(1.5.is_a? _fill_in_object_).to be true
+      expect(1.5.is_a? Float).to be true
     end
   end
 
   context "the following are Array methods" do
     it "adds a thing to the end of the method" do
-      expect([]._fill_in_method_here_(1)).to eq [1]
+      expect([].append(1)).to eq [1]
     end
 
     it "removes an item from the end of an array" do
-      expect([1]._fill_in_method_here_).to eq 1
+      expect([1].pop).to eq 1
     end
 
     it "adds an item to the front of an array" do
-      expect([1]._fill_in_method_here_("banana")).to eq ["banana", 1]
+      expect([1].unshift("banana")).to eq ["banana", 1]
     end
 
     it "removes an item from the front of an array" do
-      expect([1, "banana"]._fill_in_method_here_).to eq 1
+      expect([1, "banana"].shift).to eq 1
     end
   end
 
   context "the following are Hash methods" do
     it "adds a key and value to a Hash" do
       a_hash = {}
-      fail "remove this entire line, implement your solution here" 
+      a_hash[:jeff] = "Is cool"
       expect(a_hash.empty?).to be false 
     end
 
     it "returns a value from the hash for the given key" do
       a_hash = {magic: :johnson, shirley: :temple, "babe" => "ruth"}
 
-      expect(a_hash.fetch(:magic)).to eq _fill_in_sym_or_str 
-      expect(a_hash[:shirley]).to eq _fill_in_sym_or_str 
-      expect(a_hash["babe"]).to eq _fill_in_sym_or_str 
+      expect(a_hash.fetch(:magic)).to eq :johnson
+      expect(a_hash[:shirley]).to eq :temple
+      expect(a_hash["babe"]).to eq "ruth"
     end
 
     it "removes a key value pair from a hash" do
       a_hash = { frank: :sinatra }
-      a_hash._fill_in_method_here_(:frank)
+      a_hash.delete(:frank)
       expect(a_hash.empty?).to be true
     end
   end
 
   context "on loops, yeah!!!" do
     it "should loop over the array and return a new array" do
       loopy = [1,2,3]
-      expect(loopy._fill_in_method_here_ { |n| n + 1 }).to eq [2,3,4]
+      expect(loopy.each { |n| n + 1 }).to eq [2,3,4]
```

Comment:
Double check this one, I'm surprised it is passing.  Notice the latter half of the string: `and return a new array`

Check out the differences between `each` and a method like `map` 
- What do they take?
- What do they return?

Otherwise, nice work!

Reply:
Good call! Looks like the `each` method lets us do something with each element of an array without actually changing it, like a `for each`, while returning the original array. Wheras map returns a new mutated array! And the test wasn't passing, but now it is 😃